packages/cra-template/scripts/test.ts

```diff
@@ -1,47 +1,97 @@
 
-// TODO this should ideally be merged with `packages/fluetui/project-tests/src/createReactApp.ts`
+import {
+  addResolutionPathsForProjectPackages,
+  packProjectPackages,
+  performBrowserTest,
+  createTempDir,
+  log,
+  shEcho,
+  prepareCreateReactApp,
+} from '@fluentui/scripts/projects-test/index';
+import { findGitRoot } from '@fluentui/scripts/monorepo/index';
 
-// Clean up created files/folders on exit, even after exceptions
-// (will not catch SIGINT on windows)
-tmp.setGracefulCleanup();
-
-async function shEcho(cmd: string, cwd?: string) {
-  console.log(`+ cd ${cwd ?? '.'} && ${cmd}`);
-  await sh(cmd, cwd);
-}
+// This test is sort of like `packages/fluentui/projects-test/src/createReactApp.ts`, but it uses
+// a custom local template rather than making a generic TS project and adding our project as a dep.
+// So they share some logic, but can't be completely merged (and this test shouldn't go under
+// projects-test because it needs to run under different scoped build conditions).
 
+/**
+ * Ensure the template in this branch is requesting the version of @fluentui/react from this branch.
+ * This will probably only be an issue if there's a major version bump.
+ */
 function verifyVersion() {
-  const templateJson = JSON.parse(fs.readFileSync(path.resolve(__dirname, '../template.json'), 'utf8'));
+  const templateJson = fs.readJSONSync(path.resolve(__dirname, '../template.json'));
   const templateVersion = templateJson.package.dependencies['@fluentui/react'];
-  const reactPackageJson = JSON.parse(fs.readFileSync(path.resolve(__dirname, '../../react/package.json'), 'utf8'));
+  const reactPackageJson = fs.readJSONSync(path.resolve(__dirname, '../../react/package.json'));
   const actualVersion = reactPackageJson.version;
   if (!semver.satisfies(actualVersion, templateVersion)) {
     console.error(
-      `@fluentui/react version ${reactPackageJson} does not satisfy template version range ${templateVersion}.`,
+      `@fluentui/react version ${actualVersion} does not satisfy template version range ${templateVersion}.`,
     );
     console.error('Please update the @fluentui/react version listed in cra-template/template.json.');
     process.exit(1);
   }
 }
 
-async function runE2ETest() {
-  try {
-    const tmpDir = tmp.dirSync({ prefix: 'cra-test-', unsafeCleanup: true });
-    const craPath = path.resolve(tmpDir.name, 'cra-app');
-    await shEcho(`npx create-react-app ${craPath} --template file:.`);
-    await shEcho('CI=true yarn build', craPath);
-    await shEcho('CI=true yarn test', craPath);
-  } catch (e) {
-    console.log('@fluentui/cra-template: tests failed with error:');
-    console.log(e);
-    process.exit(1);
+/**
+ * If we tested the template as-is, it would install the latest packages from npm, which is pointless.
+ * Instead, pack up the locally-built packages and make a copy of the template which references them.
+ */
+async function prepareTemplate(logger: Function, tmpDirectory: string) {
```

Comment:
from my experience  it is not pointless. it was designed to be a proper e2e test as is.

how much seconds will this save as a tax for additional complexity and source code to maintain ?

Reply:
> from my experience it is not pointless. it was designed to be a proper e2e test as is.

Not sure what you mean by this. The purpose of the test is ensure that latest version of the template and `@fluentui/react`, *including* any local modifications, work properly together. Previously the test didn't cover local modifications to `@fluentui/react`, which seems like a major oversight. Also, this is basically the same approach that northstar is already taking in their CRA test.